pulpcore/cli/common/generic.py

```diff
@@ -435,3 +456,155 @@ def label_show(entity_ctx: PulpEntityContext, key: str) -> None:
         label_group.add_command(subcmd)
 
     return label_group
+
+
+def repository_content_command(**kwargs: Any) -> click.Group:
+    """A factory that creates a repository content command group."""
+
+    if "name" not in kwargs:
+        kwargs["name"] = "content"
+    content_contexts = kwargs.pop("contexts", {})
+    content_names = list(content_contexts.keys())
+    decorators = kwargs.pop("decorators", [href_option])
+
+    @click.group(**kwargs)
+    @click.option(
+        "--type",
+        "-t",
+        "content_type",
+        type=click.Choice(content_names, case_sensitive=False),
+        default=content_names[0],
+    )
+    @pass_pulp_context
+    @click.pass_context
+    def content_group(ctx: click.Context, pulp_ctx: PulpContext, content_type: str) -> None:
+        ctx.obj = content_contexts[content_type](pulp_ctx)
+        subcommand = ctx.command.commands[ctx.invoked_subcommand]  # type: ignore
+        if ctx.invoked_subcommand in ["add", "remove"]:
+            for field in ctx.obj.UNIQUE_LOOKUPS:
+                click.option(
+                    f"--{field.replace('_', '-')}",
+                    help=_(field.capitalize() + " of the {entity} to add to the repository"),
+                    cls=PulpOption,
+                )(subcommand)
+        elif ctx.invoked_subcommand == "modify":
+            lookup_fields = ctx.obj.UNIQUE_LOOKUPS
+            click.option(
+                "--add-content",
+                callback=load_json_callback,
+                help=_(
+                    "JSON string with list of objects to add to the repository. "
+                    "Objects can be pulp_hrefs or dictionaries containing the keys: {}".format(
+                        ", ".join(lookup_fields)
+                    )
+                ),
+            )(subcommand)
+            click.option(
+                "--remove-content",
+                callback=load_json_callback,
+                help=_(
+                    "JSON string with list of objects to remove to the repository. "
+                    "Objects can be pulp_hrefs or dictionaries containing the keys: {}".format(
+                        ", ".join(lookup_fields)
+                    )
+                ),
+            )(subcommand)
+
+    @content_group.command(name="list")
+    @click.option(
+        "--all-types",
+        is_flag=True,
+        help=_("List all content regardless of type in this repository"),
+    )
```

Comment:
Yeah, but how else would you want it to be specified? If a plugin has multiple content types this allows them to view content of just one type if they want. I could change this `all-types` field to use a for loop over the content contexts of the plugin instead of using the generic content context list call.

Reply:
I know, this might need even more deep inspection, but this basically says that the command to show all content_types is:
`pulp file repository [--type file] content --type file list --all-types` while you could expect it to be
`pulp file repository [--type file] content list`.
Can we maybe, as a middle ground have a way to specify
`pulp file repository [--type file] content --type all list` ?

Reply:
If all mapped to `pulpcore.cli.common.contexts.PulpContentContext`, would that do the trick?

Reply:
Sure I can do that.